### ISSUE TITLE
[Bug] Coredump for OCRSearch feature extraction

### ISSUE THREAD

**Issue by sauterl** [vitrivr/cineast#359]
In case others have this issue as well, I document it here:

Using the OCRSearch feature in an extraction configured as follows:

```json
{
	"input":{
		"path": "path/to/videos/",
		"depth": 1,
		"skip": 0,
		"id": {
			"name": "FileNameObjectIdGenerator",
			"properties": {}
		}
	},
	"extractors":[
		{"name": "OCRSearch"}
	],
	"metadata": [
		{"name": "TechnicalVideoMetadataExtractor"},
		{"name": "EXIFMetadataExtractor"}
	],
	"exporters":[
		{
			"name": "ShotThumbnailsExporter",
			"properties": {
				"destination":"thumbnails/"
			}
		}
	],
	"segmenter": {
		"name":"org.vitrivr.cineast.core.extraction.segmenter.video.V3CMSBSegmenter",
		"properties": {
			"folder": "path/to/msb"
		}
	},
	"database": {
		"writer": "JSON",
		"selector": "NONE",
		"host":"./ocr"
	},
	"pipeline":{
		"shotQueueSize": 20,
		"threadPoolSize": 20,
		"taskQueueSize": 20
	}
}
```
after roughly 7000 segments, a coredump stopped the extraction, which was executed using:  
``` 
java -Xmx32G -Xms32G -jar cineast-runtime/build/libs/cineast-runtime- ../cineast.json extract -e ../extraction-most.json`
``` 
This occurred on a Ubuntu 20.04.5 LTS, using openjdk
```
openjdk version "17.0.5" 2022-10-18
OpenJDK Runtime Environment (build 17.0.5+8-Ubuntu-2ubuntu120.04)
OpenJDK 64-Bit Server VM (build 17.0.5+8-Ubuntu-2ubuntu120.04, mixed mode, sharing)
```
with 40 cores of type Intel(R) Xeon(R) Silver 4210 CPU @ 2.20GHz

GPU:
NVIDIA GeForce RTX 2080 TI, nvidia-driver 450.51.06 and CUDA 11.0.228

**Comment by lucaro** [comment]
and what did the core dump say?

**Comment by sauterl** [comment]
Unfortunately there is no further information than a generic core dumped message. I did check in the user's home or the working direktory and none of the usual error logs appeared. My assumption (since a proper java coredump would produce a log file), that some native library had an issue.

**Comment by lucaro** [comment]
Yes, that's usually the only way to trigger a core dump. It should still dump something though. If it were killed by the OS before it was able to do so, you would get a different message.

**Comment by lucaro** [comment]
If this is indeed the problem, possible workaround would include:

1. making the extraction thread pool size smaller in order to reduce the number of feature instances operating simultaneously 
2. reducing the video resolution in the decoder settings

Can you check if one or both of these measures resolves your issue?

**Comment by lucaro** [comment]
This is expected behavior in this case and what you would want to happen. This only tells you that the feature extraction is slower than the decoder, and if you have more compute resources, you could increase throughput. In case you are compute - or in this case memory - limited, you'd want the pipeline to slow down rather than trying to consume more resources than are available. Were you able to run your extraction without anything crashing?

**Comment by lucaro** [comment]
Whatever works 😉 
The question cannot be answered as posed since the answer also depends on the content to be processed. Is 640x480 enough for all the text to be readable? If yes, can you go lower and still have it readable? If not, how much larger do you need to go? How long is the mean shot of your content? Does it have a lot of short sequences, or is it more of one continuous recording?
The limiting factor you have here is the longest shot you need to keep in memory, so the total maximum memory consumption is proportional to the duration of the longest shot times the video resolution. As long as you can keep this below your hardware limit, you should not experience any problems.
